### PR TITLE
Added test for adding Title and Books

### LINKED ISSUE #39: Setup tests for creating titles and books.
## Description
Use E2E with Cypress testing

## DoD:
- Test which creates title and books with cypress.
- Add test routes on backend for login.

## Context:
Talk to @itggot-TE4/red-team

### PR DESCRIPTION
**Pull request by te4-Leo-Wahlandt**
## What issue did you implement or fix?
Fixed #39 

## Summary
- You can now test if you can add a title and Book

## Type of change
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected) Added ID to form.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## How can this implementation be improved?
- Being to login or have a google session so you can add new book/s

### CODE REVIEW

File: .github/unit.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      with:
+        node-version: '12.x'
+    - name: Install packages
+      run: |
+        cd frontend
+        yarn install
+    - name: Run linter
+      run: |
+        cd frontend
+        yarn lintnf
```
Comment (TE4-jacob-lundberg) [inline .github/unit.yml]: Didn't we already have a unit.yml file with even more actions such as backend-unit-tests and api-tests?

File: frontend/cypress/integration/createTitleAndBooks.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { selector } from "postcss-selector-parser"
```
Comment (te4-david-petersson) [inline frontend/cypress/integration/createTitleAndBooks.js]: you need?

File: frontend/src/components/CreateBookFormComponent.vue [hunk 3]
```diff
@@ -5,6 +5,7 @@
         v-model="form.barcode"
         label="Barcode"
         autocomplete="off"
+        id="streckkod"
```
Comment (TE4-jacob-lundberg) [inline frontend/src/components/CreateBookFormComponent.vue]: Is it better to use id or data-attributes? Currently there is no indication on what this id is for as it is normally used as a css identifier, right? There are more occurrences in this file.

File: frontend/src/components/DropDownType.vue [hunk 4]
```diff
@@ -3,8 +3,8 @@
         b-dropdown#ddown1.m-md-2(text='Ändra Typ')
             b-dropdown-item(@click='changeToLoan') Lån
             b-dropdown-item(@click='changeToCig') ID Kort
-            b-dropdown-item(@click='changeToBook') Bok
-            b-dropdown-item(@click='changeToTitel') Titel
+            b-dropdown-item(@click='changeToBook' id='book') Bok
+            b-dropdown-item(@click='changeToTitel' id='title') Titel
```
Comment (TE4-jacob-lundberg) [inline frontend/src/components/DropDownType.vue]: Same as above comment

File: frontend/cypress/integration/createTitleAndBooks.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  // it('Login with Google'), () => {
+  //   cy.get('[id=not_signed_inow8rarn26wvu]')
+  // }
+
+  it('Open up the title', () => {
+    cy.get('[id=ddown1__BV_toggle_]').click()
+    cy.get('[id=title]').click()
+  })
+
+  it('Creates a titel', () => {
```
Comment (te4-david-petersson) [inline frontend/cypress/integration/createTitleAndBooks.js]: slight misspelling of title

File: frontend/cypress/integration/createTitleAndBooks.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  })
+    
+  it('Open up the book', () => {
+    cy.get('[id=ddown1__BV_toggle_]').click()
+    cy.get('[id=book]').click()
+  })
+
+  it('Creates a book', () => {
+    cy.get('[id=titlenamn]').type('Book made from Test')
+    cy.get('[id=streckkod]').type('232133131')
```
Comment (te4-david-petersson) [inline frontend/cypress/integration/createTitleAndBooks.js]: Consider replacing streckkod with it's english counterpart, perhaps "barcode"

File: frontend/cypress/integration/createTitleAndBooks.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    cy.get('[placeholder=isbn]').type('420-420 {enter}')
+  })
+    
+  it('Open up the book', () => {
+    cy.get('[id=ddown1__BV_toggle_]').click()
+    cy.get('[id=book]').click()
+  })
+
+  it('Creates a book', () => {
+    cy.get('[id=titlenamn]').type('Book made from Test')
```
Comment (te4-david-petersson) [inline frontend/cypress/integration/createTitleAndBooks.js]: A little confusing mixing english and swedish in the same name, perhaps "titlename" would be a better name

File: .github/unit.yml (hunk 1 above)
Comment (TE4-jacob-lundberg) [reply .github/unit.yml]: This is also duplication as there is already a unit.yml file in .github/workflows

File: .github/unit.yml [hunk 8]
```diff
@@ -0,0 +1,76 @@
+name: Unit
```
Comment (TE4-jacob-lundberg) [inline .github/unit.yml]: This file should still be removed as there already is one in .github/workflows.

File: frontend/cypress/integration/createTitleAndBooks.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
+describe('This test will create a title and a book', () => {
+  it('Opens the website', () => {
+    cy.visit('http://localhost:8080/add')
+  })
+
+  // it('Login with Google'), () => {
+  //   cy.get('[id=not_signed_inow8rarn26wvu]')
+  // }
+
+  it('Open up the title', () => {
```
Comment (TE4-jacob-lundberg) [inline frontend/cypress/integration/createTitleAndBooks.js]: Open a title? Perhaps there is a better name for this test

File: frontend/cypress/integration/createTitleAndBooks.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  })
+
+  it('Creates a title', () => {
+    cy.get('[placeholder=Namn]').type('Book made from Test')
+    cy.get('[placeholder=Kostnad]').type('2912')
+    cy.get('select').select('Kurslitteratur')
+    cy.get('[placeholder=isbn]').type('420-420 {enter}')
+  })
+    
+  it('Open up the book', () => {
```
Comment (TE4-jacob-lundberg) [inline frontend/cypress/integration/createTitleAndBooks.js]: Can a user open and read a book? Perhaps a different name would better describe the test.

**Comment by te4-Leo-Wahlandt** [comment]
David will continue with this issue while i'm gone, I'll be back on monday!

File: backend/yarn.lock [hunk 11]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (TE4-jacob-lundberg) [inline backend/yarn.lock]: Why is there a yarn.lock in backend? We use bundler for package managing, right?

File: frontend/src/components/TitelForm.vue [hunk 12]
```diff
@@ -0,0 +1,103 @@
+<template>
```
Comment (TE4-jacob-lundberg) [inline frontend/src/components/TitelForm.vue]: It looks to me like you created this file from scratch, when doing so make sure you follow the code standard specified in the wiki. I'm especially referring to naming files in this case.

Comment (te4-Leo-Wahlandt) [reply frontend/src/components/TitelForm.vue]: Isn't titelform an already existing file from develop? I don't remember adding a new file but maybe David did

**Comment by itggot-linus-styren** [comment]
@te4-david-petersson @te4-Leo-Wahlandt any update on this PR?

**Pull request closed without merging**